### PR TITLE
CASSANDRA-17992: Upgrade Netty to 4.1.96

### PR DESCRIPTION
**Pull request by jacek-lewandowski**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: test/distributed/org/apache/cassandra/distributed/test/InternodeEncryptionOptionsTest.java [hunk 1]
```diff
@@ -236,13 +236,14 @@ public void negotiatedProtocolMustBeAcceptedProtocolTest() throws Throwable
...
                               .put("internode_encryption", "all")
-                              .put("accepted_protocols", ImmutableList.of("TLSv1.1", "TLSv1.2"))
+                              .put("accepted_protocols", ImmutableList.of("TLSv1.1", "TLSv1.2", "TLSv1.3"))
                               .build());
         }).start())
         {
             InetAddress address = cluster.get(1).config().broadcastAddress().getAddress();
             int port = cluster.get(1).config().broadcastAddress().getPort();
 
+            // deprecated
```
Comment (bereng) [inline test/distributed/org/apache/cassandra/distributed/test/InternodeEncryptionOptionsTest.java:246]: Add a 'since'?

File: src/java/org/apache/cassandra/security/ThreadAwareSecurityManager.java [hunk 2]
```diff
@@ -46,6 +46,7 @@
  * This is better than the penalty of 1 to 3 percent using a standard {@code SecurityManager} with an <i>allow all</i> policy.
  * </p>
  */
+@SuppressWarnings("removal")
```
Comment (ekaterinadimitrova2) [inline src/java/org/apache/cassandra/security/ThreadAwareSecurityManager.java]: This change clears warnings and hides a future JDK removal we need to address which will probably require some significant work around UDFs, for example.
We need to have at least an open ticket for visibility and awareness if we are going to do this suppression (which I do not fully agree with, honestly).

File: test/distributed/org/apache/cassandra/distributed/test/InternodeEncryptionOptionsTest.java [hunk 3]
```diff
@@ -236,13 +236,14 @@ public void negotiatedProtocolMustBeAcceptedProtocolTest() throws Throwable
             c.set("server_encryption_options",
                   ImmutableMap.builder().putAll(validKeystore)
                               .put("internode_encryption", "all")
-                              .put("accepted_protocols", ImmutableList.of("TLSv1.1", "TLSv1.2"))
+                              .put("accepted_protocols", ImmutableList.of("TLSv1.1", "TLSv1.2", "TLSv1.3"))
```
Comment (ekaterinadimitrova2) [inline test/distributed/org/apache/cassandra/distributed/test/InternodeEncryptionOptionsTest.java:239]: Good call, thanks!

File: test/unit/org/apache/cassandra/cql3/RandomSchemaTest.java [hunk 4]
```diff
@@ -128,18 +128,18 @@ public void test()
                 {
                     ByteBuffer[] partitionKeys = Arrays.copyOf(expected, partitionColumnCount);
                     ByteBuffer[] rowKey = Arrays.copyOf(expected, primaryColumnCount);
-                    execute(insertStmt, expected);
+                    execute(insertStmt, (Object[]) expected);
```
Comment (ekaterinadimitrova2) [inline test/unit/org/apache/cassandra/cql3/RandomSchemaTest.java]: In general I prefer not to address unrelated issues (in this case warnings) at random tickets. In this case I propose we keep them but in a separate commit, mentioned also on the ticket.

File: conf/jvm17-server.options [hunk 5]
```diff
@@ -130,4 +130,8 @@
 # inferior performance and risks exceeding MaxDirectMemory
 -Dio.netty.tryReflectionSetAccessible=true
 
+# Revert changes in defaults introduced in https://netty.io/news/2022/03/10/4-1-75-Final.html
+-Dio.netty.allocator.useCacheForAllThreads=true
+-Dio.netty.allocator.maxOrder=11
+
```
Comment (ekaterinadimitrova2) [inline conf/jvm17-server.options:136]: Good call, maybe we also open a follow up ticket to explore if we can benefit of changing anything on Cassandra side. too? My understanding is that will require at a minimum some performance testing.

File: .build/parent-pom-template.xml [hunk 6]
```diff
@@ -676,19 +682,35 @@
...
-        <version>4.1.58.Final</version>
+        <version>4.1.94.Final</version>
         <type>pom</type>
         <scope>provided</scope>
       </dependency>
       <dependency>
         <groupId>io.netty</groupId>
         <artifactId>netty-all</artifactId>
-        <version>4.1.58.Final</version>
+        <version>4.1.94.Final</version>
```
Comment (ekaterinadimitrova2) [inline .build/parent-pom-template.xml]: There is already 4.1.96

File: src/java/org/apache/cassandra/security/ThreadAwareSecurityManager.java (hunk 2 above)
Comment (jacek-lewandowski) [reply src/java/org/apache/cassandra/security/ThreadAwareSecurityManager.java]: Ok, I'll remove it
